### PR TITLE
Freebayes version update 1.3.5

### PR DESCRIPTION
**Pull request by dorton21**
The freebayes devs switched build packages from Makefile to Meson.

### CODE REVIEW

File: var/spack/repos/builtin/packages/freebayes/package.py [hunk 1]
```diff
@@ -6,27 +6,53 @@
...
 
+    # Deps for @1.3.5
+    depends_on('htslib', when='@1.3.5')
+    depends_on('zlib', type=('build', 'run'), when='@1.3.5')
+    depends_on('xz', type=('build', 'run'), when='@1.3.5')
+    depends_on('parallel', type=('build', 'run'), when='@1.3.5')
+    depends_on('ninja', type=('build', 'run'), when='@1.3.5')
+    depends_on('vcftools', type=('build', 'run'), when='@1.3.5')
+    depends_on('bc', type=('build', 'run'), when='@1.3.5')
+    depends_on('samtools', type=('build', 'run'), when='@1.3.5')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/freebayes/package.py]: Can we use an open range in the conditions:
```suggestion
    # Deps for @1.3.5
    depends_on('htslib', when='@1.3.5:')
    depends_on('zlib', type=('build', 'run'), when='@1.3.5:')
    depends_on('xz', type=('build', 'run'), when='@1.3.5:')
    depends_on('parallel', type=('build', 'run'), when='@1.3.5:')
    depends_on('ninja', type=('build', 'run'), when='@1.3.5:')
    depends_on('vcftools', type=('build', 'run'), when='@1.3.5:')
    depends_on('bc', type=('build', 'run'), when='@1.3.5:')
    depends_on('samtools', type=('build', 'run'), when='@1.3.5:')
```
?

File: var/spack/repos/builtin/packages/freebayes/package.py [hunk 2]
```diff
@@ -6,27 +6,53 @@
...
-class Freebayes(MakefilePackage):
+class Freebayes(MesonPackage):
     """Bayesian haplotype-based genetic polymorphism discovery and
        genotyping."""
 
     homepage = "https://github.com/ekg/freebayes"
+    url      = "https://github.com/freebayes/freebayes/releases/download/v1.3.5/freebayes-1.3.5-src.tar.gz"
     git      = "https://github.com/ekg/freebayes.git"
 
+    version('1.3.5', sha256='7e2635690e916ed85cec36b3263e6e5357413a4f2bf3035362d9749335e8a696')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/freebayes/package.py:17]: If it makes sense you can also add `deprecated=True` to v1.1.0

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/freebayes/package.py [hunk 3]
```diff
@@ -6,27 +6,53 @@
...
 
+    # Deps for @1.3.5 and beyond
+    depends_on('htslib', when='@1.3.5:')
+    depends_on('zlib', type=('build', 'run'), when='@1.3.5:')
+    depends_on('xz', type=('build', 'run'), when='@1.3.5:')
+    depends_on('parallel', type=('build', 'run'), when='@1.3.5:')
+    depends_on('ninja', type=('build', 'run'), when='@1.3.5:')
+    depends_on('vcftools', type=('build', 'run'), when='@1.3.5:')
+    depends_on('bc', type=('build', 'run'), when='@1.3.5:')
+    depends_on('samtools', type=('build', 'run'), when='@1.3.5:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/freebayes/package.py]: Many of these have no `prefix.bin` directory, so `type=('build', 'run')` won't do anything. I think you want to leave it as the default deptype. `ninja` might be `type='build'`. You'll have to see how the build system actually uses these tools (whether or not it links to them).

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/freebayes/package.py [hunk 4]
```diff
@@ -6,27 +6,53 @@
...
+    depends_on('zlib', when='@1.3.5:')
+    depends_on('xz', when='@1.3.5:')
+    depends_on('parallel', when='@1.3.5:')
+    depends_on('vcftools', when='@1.3.5:')
+    depends_on('bc', when='@1.3.5:')
+    depends_on('samtools', when='@1.3.5:')
+
     parallel = False
 
+    @when('@1.1.0:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/freebayes/package.py]: Should this be `@:1.1.0`? We're now using Meson to generate the Makefile, right?

**Review by adamjstewart — APPROVED** [review]

**Review by alalazo — APPROVED** [review]

**Pull request merged**